packages/platforms/electron-player/src/main/api/file-system.ts

```diff
@@ -0,0 +1,244 @@
+  StoreFileReturnValue,
+} from '@castmill/cache';
+
+import { createHash } from 'crypto';
+import { writeFile, mkdir, readdir, unlink, stat, rename } from 'fs/promises';
+import { createWriteStream } from 'fs';
+import { join } from 'path';
+import { URL } from 'url';
+import { net } from 'electron';
+import os from 'os';
+
+// Only works with CommonJS require
+const checkDiskSpace = require('check-disk-space').default;
+
+console.log('checkDiskSpace', checkDiskSpace);
+// Get home directory
+const homeDir = os.homedir();
+
+// Base directory for storage
+const BASE_DIR = join(homeDir, 'castmill-electron-file-storage');
+
+/*
+ * Initialize storage
+ */
+export async function initStorage(storagePath: string): Promise<void> {
+  const fullPath = join(BASE_DIR, storagePath);
+  try {
+    await mkdir(fullPath, { recursive: true });
+  } catch (error) {
+    console.error('Failed to initialize storage:', error);
+    throw error;
+  }
+}
+
+/*
+ * Get storage info. Returns the used and total available storage space.
+ * We never use more than 50% of the total available space to leave space
+ * for updates etc.
+ */
+export async function getStorageInfo(
+  storagePath: string
+): Promise<StorageInfo> {
+  const fullPath = join(BASE_DIR, storagePath);
+  try {
+    const files = await readdir(fullPath);
+    let used = 0;
+    for (const file of files) {
+      const stats = await stat(join(fullPath, file));
+      used += stats.size;
+    }
+    const total = (await getFreeDiskSpace(fullPath)) + used * 0.5;
+    return { used, total };
+  } catch (error) {
+    console.error('Failed to get storage info:', error);
+    throw error;
+  }
+}
+
+/*
+ * List all files in the storage
+ */
+export async function listFiles(storagePath: string): Promise<StorageItem[]> {
+  const fullPath = join(BASE_DIR, storagePath);
+  try {
+    const files = await readdir(fullPath);
+    return Promise.all(
+      files
+        .filter((file) => !file.endsWith('.tmp')) // Exclude temporary files
+        .map(async (file) => {
+          const filePath = join(fullPath, file);
+          const stats = await stat(filePath);
+          return {
+            url: filePath,
+            size: stats.size,
+          };
+        })
+    );
+  } catch (error) {
+    console.error('Failed to list files:', error);
+    throw error;
+  }
+}
+
+/*
+ * Store a file in the storage. If data is provided, it is used as the file
+ * content. Otherwise, the file is downloaded from the URL.
+ */
+export async function storeFile(
+  storagePath: string,
+  url: string,
+  data = null
+): Promise<StoreFileReturnValue> {
+  // Must be imported dynamically since the main process is a CommonJS module but @castmill/cache is an ES module
+  const { StoreResult } = await import('@castmill/cache');
+
+  const fullPath = join(BASE_DIR, storagePath);
+  try {
+    const filename = getFileName(url);
+
+    const filePath = join(fullPath, filename);
+    const tempPath = filePath + '.tmp';
```

Comment:
This is not important right now, but in the future it would be better to have the tmp directory as a configurable path, as in systems like linux for example there are special directories that get cleaned automatically for this exact purpose.